MSAL/src/MSALPublicClientApplication.m

```diff
@@ -174,140 +202,124 @@ - (id)initWithClientId:(NSString *)clientId
 
 #endif
 
-- (id)initWithClientId:(NSString *)clientId
-         keychainGroup:(NSString *)keychainGroup
-             authority:(MSALAuthority *)authority
-           redirectUri:(NSString *)redirectUri
-                 error:(NSError * __autoreleasing *)error
+- (nullable instancetype)initWithConfiguration:(nonnull MSALPublicClientApplicationConfig *)config
+                                         error:(NSError * _Nullable __autoreleasing * _Nullable)error
 {
     if (!(self = [super init]))
     {
         return nil;
     }
-
-    if ([NSString msidIsStringNilOrBlank:clientId])
+    
+    // Verify required fields
+    if ([NSString msidIsStringNilOrBlank:config.clientId])
     {
         MSAL_ERROR_PARAM(nil, MSALErrorInvalidParameter, @"clientId is a required parameter and must not be nil or empty.");
         return nil;
     }
 
-    _clientId = clientId;
-
     NSError *msidError = nil;
-    
-    if (authority)
-    {
-        _authority = authority;
-    }
-    else
-    {
-        // TODO: Rationalize our default authority behavior (#93)
-        NSURL *authorityURL = [NSURL URLWithString:s_defaultAuthorityUrlString];
-        _authority = [[MSALAADAuthority alloc] initWithURL:authorityURL context:nil error:&msidError];
-    }
-
-    if (!_authority)
-    {
-        if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
-        return nil;
-    }
-
-    MSALRedirectUri *msalRedirectUri = [MSALRedirectUriVerifier msalRedirectUriWithCustomUri:redirectUri
-                                                                                    clientId:clientId
+    MSALRedirectUri *msalRedirectUri = [MSALRedirectUriVerifier msalRedirectUriWithCustomUri:config.redirecrUri
+                                                                                    clientId:config.clientId
                                                                                        error:&msidError];
-
+    
     if (!msalRedirectUri)
     {
         if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
         return nil;
     }
-
-    _redirectUri = msalRedirectUri;
-
+    
+    config.verifiedRedirectUri = msalRedirectUri;
+    
 #if TARGET_OS_IPHONE
     // Optional Paramater
-    _keychainGroup = keychainGroup;
-
-    MSIDKeychainTokenCache *dataSource;
-    if (_keychainGroup == nil)
-    {
-        _keychainGroup = [[NSBundle mainBundle] bundleIdentifier];
-    }
+    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:MSALGlobalConfig.cacheConfig.keychainSharingGroup];
```

Comment:
nit: you're using self.keychainGroup and MSALGlobalConfig.cacheConfig.keychainSharingGroup interchangeably. Can we use one consistently?

Reply:
sure :p